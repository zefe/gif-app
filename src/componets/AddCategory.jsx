import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChage = (e) => {
        console.log(e.target.value)
        setInputValue( e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 2){
            /* no se tiene acceso a las categorias porque no lo paso como argumento, 
            con solo mandar la referencia al setCategories que puede llamar con un callback que tiene el estado anterior*/
            setCategories((category) => [ inputValue, ...category]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input
                type="text"
                value={ inputValue }
                onChange={handleInputChage}
                placeholder="search"
            />

        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
