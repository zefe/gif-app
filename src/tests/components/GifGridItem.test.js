import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../componets/GifGridItem';

describe('Test in <GifGridItem />', () => {

    const title = 'One Title';
    const url = 'https://localhost/alto.jpg';
    const wrapper = shallow( <GifGridItem title={title} url={url} /> )

    test('should be the component conrectly', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('should be a paragraph with title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
        
    })

    test('should be the img with src and alt props of component', () => {
        const img = wrapper.find('img');
        expect( img.prop('src')).toBe( url );
        expect( img.prop('alt')).toBe( title );
    })

    test('should have animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn').toBe(true))
    })
    
    

 
    
    
    
})