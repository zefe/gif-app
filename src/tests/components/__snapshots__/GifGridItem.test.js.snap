// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem /> should be the component conrectly 1`] = `
<div
  className="card animate__animated animate__bounce "
>
  <img
    alt="One Title"
    src="https://localhost/alto.jpg"
  />
  <p>
    One Title
     
  </p>
</div>
`;
